content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,293 @@
+---
+kind: tutorial
+languages: [javascript]
+sidebar: "language:javascript"
+prev: /languages/javascript/codewars-test/
+---
+
+# JavaScript: creating and translating a kata
+
+This article is intended for kata authors and translators who would like to create new content in JavaScript. It attempts to explain how to create and organize things in a way conforming to [authoring guidelines](/authoring/guidelines/), shows the most common pitfalls and how to avoid them.
+
+This article is not a standalone tutorial on creating kata or translations. It's meant to be a complementary, JavaScript-specific part of a more general set of HOWTOs and guidelines related to [content authoring](/authoring/). If you are going to create a JavaScript translation, or a new JavaScript kata from scratch, please make yourself familiar with the aforementioned documents related to authoring in general first. 
+
+
+## General info
+
+Any technical information related to the JavaScript setup on Codewars can be found on the [JavaScript reference](/languages/javascript/) page (language versions, available libraries, and setup of the code runner).
+
+## Description
+
+JavaScript code blocks can be inserted with JavaScript-specific part in [sequential code blocks](/references/markdown/extensions/#sequential-code-blocks):
+
+~~~
+```javascript
+
+...your code here...
+
+```
+~~~
+
+JavaScript-specific paragraphs can be inserted with [language conditional rendering](/references/markdown/extensions/#conditional-rendering):
+
+```
+~~~if:javascript
+
+...text visible only for JavaScript description...
+
+~~~
+
+~~~if-not:javascript
+
+...text not visible in JavaScript description...
+
+~~~
+```
+
+## Tasks and Requirements
+
+Some concepts don't always translate well to or from JavaScript. Because of this, some constructs should be avoided and some translations just shouldn't be done.
+- Avoid returning different data types depending on the situation (_"Return the result, or the string 'Error' if no result can be found..."_). JavaScript is dynamically typed, which is not the case for some other languages. Returning `null` might be appropriate in some situations, but throwing an error might be better in others.
+
+Some kata just should not be translated into JavaScript because it can be difficult to keep their initial idea:
+- Some kata might be meant for another language specifically. For example, a kata about `Python: Learning Classes` to do something should probably not be translated to JavaScript, even though JS has classes too.
```

Comment:
```suggestion
- Some kata might be meant for another language specifically. For example, a kata about `Python: Learning Classes` should probably not be translated to JavaScript, even though JS has `class` syntax.
```

Reply:
Technically speaking, JS only has objects and the prototype chain. `class` syntax should be viewed as syntactical sugar and not an actual JS construct.